cpp/src/groupby/sort/sort_helper.cu

```diff
@@ -195,17 +187,18 @@ sort_groupby_helper::index_vector const& sort_groupby_helper::group_labels(
   if (_group_labels) return *_group_labels;
 
   // Get group labels for future use in segmented sorting
-  _group_labels = std::make_unique<index_vector>(num_keys(stream), stream);
-
-  auto& group_labels = *_group_labels;
-  if (num_keys(stream) == 0) return group_labels;
+  auto group_labels = std::make_unique<index_vector>(num_keys(stream), stream);
+
+  if (num_keys(stream)) {
+    cudf::detail::label_segments(group_offsets(stream).begin(),
+                                 group_offsets(stream).end(),
```

Comment:
I have no idea if the compiler in the future can execute these two lines in parallel in multiple threads, which in turn fully executes the `group_offsets(stream)` function twice when the `_group_offsets` vector was not generated. IMO this is better to safeguard us from such situation:
```suggestion
   auto const& offsets = group_offsets(stream);
    cudf::detail::label_segments(offsets.begin(), offsets.end(),
```

Reply:
This is the premise of the PR in general. I'm not sure if it is possible but thought it best for the code to be correct.